### PR TITLE
Azd 0.5.0 beta1

### PR DESCRIPTION
**Pull request by lechnerc77**
## Purpose

This PR contains the changes necessary to be compliant with version [0.5.0-beta.1](https://github.com/Azure/azure-dev/releases/tag/azure-dev-cli_0.5.0-beta.1) of the Azure Developer CLI.

This especially comprises the changes in the CI/CD specific area due to the change in the login procedure but also further updates and enhancements of the templates

## Does this introduce a breaking change?

```
[X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe: Fixes for azd compatibility
```

## How to Test

*  Get the code from the PR

* Test the code:
   * `azd provision` (check infrastructure setup)
   * `azd deploy` (deployment of application)
   * `azd up` (Provide infrastructure and app in one run)
   * `azd pipeline config`- configuration of pipeline

## What to Check

All operations from above should work successfully.

## Other Information

`azd login` does not work for me, see issue <https://github.com/Azure/azure-dev/issues/1366>. Not relevant for the code changes, but to be kept in mind when testing the changes.

### CODE REVIEW

**Comment by lechnerc77** [comment]
Added compatibility information to version 0.5.0-beta2 of `azd` which contains the bug fix for logging into personal Azure accounts via `azd`

File: .azdo/pipelines/azure-dev.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ pool:
 container: mcr.microsoft.com/azure-dev-cli-apps:latest
 
 steps:
+  - pwsh: |
+      azd config set auth.useAzCliAuth "true"
```
Comment (MartinPankraz) [inline .azdo/pipelines/azure-dev.yml:11]: dachte dises flag braucht es seit dem fix nicht mehr?

Comment (lechnerc77) [reply .azdo/pipelines/azure-dev.yml:11]: Good catch. That is basically a copy&paste form the original template located here: 
<https://github.com/Azure-Samples/todo-nodejs-mongo/blob/main/.azdo/pipelines/azure-dev.yml>

Maybe a azdo limitation?

Comment (lechnerc77) [reply .azdo/pipelines/azure-dev.yml:11]: And yes it is a limitation of the Azure DevOps pipeline as referenced here <https://github.com/Azure/azure-dev/issues/1126>

**Review by MartinPankraz — APPROVED** [review]
retested. All good

**Pull request merged**